src/formatDate.js

```diff
@@ -8,7 +8,36 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const SEPARATOR = fromFormat[fromFormat.length - 1];
+  const JOINER = toFormat[toFormat.length - 1];
+  const DATEPARTS = date.split(SEPARATOR);
+
+  while (DATEPARTS.length > 1 && DATEPARTS[DATEPARTS.length - 1] === '') {
+    DATEPARTS.pop();
+  }
+
+  const DATEMAP = {};
+
+  fromFormat.forEach((part, index) => {
+    DATEMAP[part] = DATEPARTS[index];
+  });
+
+  if (fromFormat.includes('YYYY') && toFormat.includes('YY')) {
+    DATEMAP['YY'] = DATEMAP['YYYY'].slice(-2);
+  }
+
+  if (fromFormat.includes('YY') && toFormat.includes('YYYY')) {
+    const FULLYEAR
+    = parseInt(DATEMAP['YY']) < 30
+      ? '20' + DATEMAP['YY']
+      : '19' + DATEMAP['YY'];
+
+    DATEMAP['YYYY'] = FULLYEAR;
+  }
+
+  const FORMATTEDDATE = toFormat.map((part) => DATEMAP[part]);
+
+  return FORMATTEDDATE.slice(0, -1).join(JOINER);
 }
```

Comment:
The function `formatDate` is missing input validation. It's important to check if the provided `date`, `fromFormat`, and `toFormat` arguments are valid and in the expected format. This includes checking if the `date` string matches the `fromFormat` pattern, and if the `fromFormat` and `toFormat` arrays have the correct length and contain valid date part identifiers. Without these checks, the function may produce incorrect results or throw errors when encountering unexpected input.